lib/usagereporter/usagereporter.go

```diff
@@ -92,6 +93,7 @@ type SubmittedEvent[T any] struct {
 }
 
 type UsageReporter[T any] struct {
+	mtx sync.RWMutex
```

Comment:
A mutex that covers a single bool and is not otherwise used to explicitly block or wait should probably be replaced by the use of an atomic bool.

(also, prefer just a regular `sync.Mutex` to a `RWMutex` unless there's a significant amount of readers and there's a significant amount of readers relative to the amount of writers)

Reply:
Not sure if switching to atomic will be totally safe because Don't we need to sync the `r.events` and make sure after the   `wasGracefulStopScheduled`  chancel  was closed we won't submit new events from https://github.com/gravitational/teleport/blob/master/lib/usagereporter/usagereporter.go#L296 ?

Reply:
Oh yeah, it's not equivalent here.